assets/gateway-api/gateway.networking.k8s.io_gatewayclasses.yaml

```diff
@@ -0,0 +1,433 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    api-approved.kubernetes.io: https://github.com/kubernetes-sigs/gateway-api/pull/1086
+    gateway.networking.k8s.io/bundle-version: v0.6.0-dev
```

Comment:
Good point.  https://github.com/maistra/istio/blob/maistra-2.4/go.mod#L106 (permalink: https://github.com/maistra/istio/blob/7b1d1a9b9be85e448ee8b235d1f5a52b8d09955b/go.mod#L106) has `sigs.k8s.io/gateway-api v0.5.1-0.20220921185115-ee7a83814203`.  https://github.com/kubernetes-sigs/gateway-api/compare/ee7a83814203...v0.6.1 has some improvements to API field descriptions, which would be nice to have.  However, the diff also shows that some status conditions definitions and defaults changed, some `additionalPrinterColumns` definitions changed (which affects oc/kubectl output), and the referencegrants CRD was promoted from experimental to standard.  It makes sense to use CRDs that match the Maistra version so that status conditions match.  If referencegrants work in Maistra 2.4, then it must be using the experimental CRDs from v0.5.1-0.20220921185115-ee7a83814203, so we should use those here.  

This also means we'll want to bump the vendored CRDs when OSSM bumps its vendored API definitions.  That's something we'll need to keep in mind.

Reply:
Fixed in https://github.com/openshift/cluster-ingress-operator/compare/c1c4ef672c6a88f64ad61f33ffaa30d83b993a44..503e0364c54c9f869b0ea8aaa7175b7bf458dca8.

Reply:
Okay make sense, and the fact reference grant will still be in under the v1alpha2 API should be in our release documentation (though documenting the version of Gateway API also might suffice).

So, at the moment, we are still using `v0.6.0-dev`. Is intentional? Is it better to align with the version that OSSM 2.4 is currently vendoring?

Reply:
Oh, facepalm, I updated manifests but not the vendored gateway-api.  I don't know that it matters, but I'll fix that just in case.

Reply:
https://github.com/openshift/cluster-ingress-operator/compare/503e0364c54c9f869b0ea8aaa7175b7bf458dca8..53c61b464207eeeee5709eab343757c1d3a10f3f fixes the vendored gateway-api package.

Reply:
Oh I actually forgot about `go.mod`'s version Gateway API, but good catch. I'm actually talking about the version `gateway.networking.k8s.io/bundle-version: v0.6.0-dev` that is still in the manifest. That isn't correct right? Still says `v0.6.0-dev`.